src/screenshot/screenshot-compare.ts

```diff
@@ -42,10 +42,10 @@ export async function compareScreenshot(
     testPath: testPath,
     width: width,
     height: height,
-    deviceScaleFactor: emulateConfig.viewport.deviceScaleFactor,
-    hasTouch: emulateConfig.viewport.hasTouch,
-    isLandscape: emulateConfig.viewport.isLandscape,
-    isMobile: emulateConfig.viewport.isMobile,
+    deviceScaleFactor: emulateConfig?.viewport?.deviceScaleFactor,
+    hasTouch: emulateConfig?.viewport?.hasTouch,
+    isLandscape: emulateConfig?.viewport?.isLandscape,
+    isMobile: emulateConfig?.viewport?.isMobile,
```

Comment:
Is the optional chaining on `emulateConfig` necessary? 
```suggestion
    deviceScaleFactor: emulateConfig.viewport?.deviceScaleFactor,
    hasTouch: emulateConfig.viewport?.hasTouch,
    isLandscape: emulateConfig.viewport?.isLandscape,
    isMobile: emulateConfig.viewport?.isMobile,
```

Reply:
FWIW I see some SNC's related to `emulateConfig.viewport`, but only bc we haven't added any optional chaining to `viewport` on lines 108-109. Running `npx tsc --strictNullChecks --noEmit` at ec613f331615f724fb39a9816b65a0930f2643da:

```
src/screenshot/screenshot-compare.ts:108:39 - error TS18048: 'emulateConfig.viewport' is possibly 'undefined'.

108       const naturalWidth = Math.round(emulateConfig.viewport.width * emulateConfig.viewport.deviceScaleFactor);
                                          ~~~~~~~~~~~~~~~~~~~~~~

src/screenshot/screenshot-compare.ts:108:70 - error TS18048: 'emulateConfig.viewport' is possibly 'undefined'.

108       const naturalWidth = Math.round(emulateConfig.viewport.width * emulateConfig.viewport.deviceScaleFactor);
                                                                         ~~~~~~~~~~~~~~~~~~~~~~

src/screenshot/screenshot-compare.ts:108:70 - error TS18048: 'emulateConfig.viewport.deviceScaleFactor' is possibly 'undefined'.

108       const naturalWidth = Math.round(emulateConfig.viewport.width * emulateConfig.viewport.deviceScaleFactor);
                                                                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/screenshot/screenshot-compare.ts:109:40 - error TS18048: 'emulateConfig.viewport' is possibly 'undefined'.

109       const naturalHeight = Math.round(emulateConfig.viewport.height * emulateConfig.viewport.deviceScaleFactor);
                                           ~~~~~~~~~~~~~~~~~~~~~~

src/screenshot/screenshot-compare.ts:109:72 - error TS18048: 'emulateConfig.viewport' is possibly 'undefined'.

109       const naturalHeight = Math.round(emulateConfig.viewport.height * emulateConfig.viewport.deviceScaleFactor);
                                                                           ~~~~~~~~~~~~~~~~~~~~~~

src/screenshot/screenshot-compare.ts:109:72 - error TS18048: 'emulateConfig.viewport.deviceScaleFactor' is possibly 'undefined'.

109       const naturalHeight = Math.round(emulateConfig.viewport.height * emulateConfig.viewport.deviceScaleFactor);
```

Reply:
those ones are a bit trickier though because `foo?.bar` will return `undefined` if the chain fails at any point, so getting rid of the error will require some default number or something along those lines. It does look like it's relatively straightforward without the codebase to make `.viewport` a required property, removing the need for a lot of these checks, but that would require changing a public interface

Reply:
Ah gotcha - nvm in that case 👍